core/src/main/scala/org/apache/spark/status/api/v1/OneStageResource.scala

```diff
@@ -0,0 +1,192 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.status.api.v1
+
+import javax.ws.rs._
+import javax.ws.rs.core.MediaType
+
+import org.apache.spark.SparkException
+import org.apache.spark.scheduler.StageInfo
+import org.apache.spark.ui.jobs.JobProgressListener
+import org.apache.spark.ui.jobs.UIData.StageUIData
+import org.apache.spark.util.SparkEnum
+
+@Produces(Array(MediaType.APPLICATION_JSON))
+private[v1] class OneStageResource(uiRoot: UIRoot) {
+
+  @GET
+  @Path("")
+  def stageData(
+    @PathParam("appId") appId: String,
+    @PathParam("stageId") stageId: Int
+  ): Seq[StageData] = {
+    forStage(appId, stageId){ (listener,stageAttempts) =>
+      stageAttempts.map { case (status, stageInfo) =>
+        val stageUiData = listener.synchronized {
+          listener.stageIdToData.get((stageInfo.stageId, stageInfo.attemptId)).
+            getOrElse(throw new SparkException("failed to get full stage data for stage: " +
+            stageInfo.stageId + ":" + stageInfo.attemptId)
+            )
+        }
+        AllStagesResource.stageUiToStageData(status, stageInfo, stageUiData,
+          includeDetails = true)
+      }
+    }
+  }
+
+  @GET
+  @Path("/{attemptId: \\d+}")
+  def oneAttemptData(
+    @PathParam("appId") appId: String,
+    @PathParam("stageId") stageId: Int,
+    @PathParam("attemptId") attemptId: Int
+  ): StageData = {
+    forStageAttempt(appId, stageId, attemptId) { case (status, stageInfo, stageUiData) =>
+      AllStagesResource.stageUiToStageData(status, stageInfo, stageUiData,
+        includeDetails = true)
+    }
+  }
+
+  @GET
+  @Path("/{attemptId: \\d+}/taskSummary")
+  def stageData(
+    @PathParam("appId") appId: String,
+    @PathParam("stageId") stageId: Int,
+    @PathParam("attemptId") attemptId: Int,
+    @DefaultValue("0.05,0.25,0.5,0.75,0.95") @QueryParam("quantiles") quantileString: String
```

Comment:
I see in certain places you're using comma-separated values, while in others you're using `List[String]`. Is that intentional?